File: AIbotAdmin.Server/Models/T_QALog.cs
using System.ComponentModel.DataAnnotations;

namespace AIbotAdmin.Server.Models
{
    public class T_QALog
    {
        public long QALogCD { get; set; }
        public string SessionId { get; set; } = string.Empty;
        public int TurnNo { get; set; }
        public long? UserCD { get; set; }
        public string QuestionText { get; set; } = string.Empty;
        public string AnswerText { get; set; } = string.Empty;
        public int? ResolvedTurnNo { get; set; }
        public DateTime? RegisteredAt { get; set; }
    }

    public class ChatSessionSummary
    {
        public string SessionId { get; set; } = string.Empty;
        public long? UserCD { get; set; }
        public string UserName { get; set; } = string.Empty;
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int TurnCount { get; set; }
        public string Status { get; set; } = string.Empty;
        public string LastQuestion { get; set; } = string.Empty;
        public string LastAnswer { get; set; } = string.Empty;
    }

    public class ChatSession
    {
        public string SessionId { get; set; } = string.Empty;
        public long? UserCD { get; set; }
        public string UserName { get; set; } = string.Empty;
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int TurnCount { get; set; }
        public string Status { get; set; } = string.Empty;
        public List<T_QALog> QALogs { get; set; } = new List<T_QALog>();
    }

    public class ChatListRequest : PagedRequest
    {
        public string? SessionId { get; set; }
        public long? UserCD { get; set; }
        public string? UserName { get; set; }
        public string? Status { get; set; }
        public DateTime? SelectedDate { get; set; }      // Ngày được chọn
        public TimeSpan? StartTime { get; set; }         // Giờ bắt đầu (HH:mm)
        public TimeSpan? EndTime { get; set; }           // Giờ kết thúc (HH:mm)
        public string? SearchText { get; set; }
        public string? SortColumn { get; set; } = "StartTime";
        public string? SortDirection { get; set; } = "DESC";
    }

    public class ChatExportRequest
    {
        public string? SessionId { get; set; }
        public long? UserCD { get; set; }
        public string? UserName { get; set; }
        public string? Status { get; set; }
        public DateTime? SelectedDate { get; set; }
        public TimeSpan? StartTime { get; set; }
        public TimeSpan? EndTime { get; set; }
        public string? SearchText { get; set; }
        public string Format { get; set; } = "csv";
    }
}

File: AIbotAdmin.client/src/types/t-qalog.ts
export interface T_QALog {
    qaLogCD: number;
    sessionId: string;
    turnNo: number;
    userCD: number | null;
    questionText: string;
    answerText: string;
    resolvedTurnNo: number | null;
    registeredAt: string; 
}

export interface ChatSession {
    sessionId: string;
    userCD: number | null;
    userName: string;
    startTime: string;
    endTime: string; 
    turnCount: number;
    status: 'completed' | 'uncompleted';
    lastQuestion: string;
    lastAnswer: string;
}

export interface ChatSessionDetail {
    sessionId: string;
    userCD: number | null;
    userName: string;
    startTime: string;
    endTime: string;
    turnCount: number;
    status: 'completed' | 'uncompleted';
    qaLogs: Array<{
        qaLogCD: number;
        sessionId: string;
        turnNo: number;
        userCD: number | null;
        questionText: string;
        answerText: string;
        resolvedTurnNo: number | null;
        registeredAt: string;
    }>;
}

export interface ChatListRequest {
    pageNumber: number;
    pageSize: number;
    sessionId?: string;
    userCD?: number;
    userName?: string;
    status?: 'completed' | 'uncompleted';
    selectedDate?: string;   // Format: YYYY-MM-DD
    startTime?: string;      // Format: HH:mm
    endTime?: string;        // Format: HH:mm
    searchText?: string;
    sortColumn?: string;
    sortDirection?: 'ASC' | 'DESC';
}

export interface ChatExportRequest {
    sessionId?: string;
    userCD?: number;
    userName?: string;
    status?: 'completed' | 'uncompleted';
    selectedDate?: string;
    startTime?: string;
    endTime?: string;
    searchText?: string;
    format?: 'csv' | 'excel' | 'pdf';
}

export interface PaginationInfo {
    pageNumber: number;
    pageSize: number;
    totalCount: number;
    totalPages: number;
    hasPrevious: boolean;
    hasNext: boolean;
}

export interface ChatSessionsResponse {
    status: 'ok' | 'error';
    message?: string;
    items: ChatSession[];
    totalCount: number;
    pageNumber: number;
    pageSize: number;
    totalPages: number;
    hasPrevious: boolean;
    hasNext: boolean;
}

export interface ChatSessionDetailResponse {
    status: 'ok' | 'error';
    message?: string;
    data: ChatSessionDetail;
}

export interface ApiErrorResponse {
    status: 'error';
    message: string;
    error?: string;
}